libgnucash/engine/gnc-commodity.cpp

```diff
@@ -2114,7 +2097,7 @@ gnc_commodity_table_add_namespace(gnc_commodity_table * table,
     if (!ns)
     {
         ns = static_cast<gnc_commodity_namespace*>(g_object_new(GNC_TYPE_COMMODITY_NAMESPACE, nullptr));
-        ns->cm_table = g_hash_table_new(g_str_hash, g_str_equal);
+        StrCommodityMap().swap(ns->cm_table);
```

Comment:
Risky. I haven't looked at the standard but [c++ reference](Swapping with a default-constructed temporary doesn't say anything about swapping to 0-filled memory running the constructor on that memory.
Better to use placement new to be sure that the default constructor is called.